src/providers/assets/assetsManager.tsx

```diff
@@ -1,117 +1,142 @@
 
         try {
           await assetTestPromiseWithTimeout
-          addAsset(contentId, { url: assetUrl })
-          removePendingAsset(contentId)
-          removeAssetBeingResolved(contentId)
+          addAsset(dataObject.id, { url: assetUrl })
+          removePendingAsset(dataObject.id)
+          removeAssetBeingResolved(dataObject.id)
           return
-        } catch (e) {
-          // ignore anything else than TimeoutError as it will be handled by assetTestPromise.catch
-          if (e instanceof TimeoutError) {
-            AssetLogger.assetTimeout(assetDetails)
-            ConsoleLogger.warn('Asset load timed out', assetDetails)
+        } catch (err) {
+          if (err instanceof TimeoutError) {
+            // AssetLogger.assetTimeout(assetDetails)
+            ConsoleLogger.warn(`Distributor didn't respond in ${ASSET_RESPONSE_TIMEOUT} seconds`, {
+              dataObject,
+              distributor,
+            })
+          } else {
+            SentryLogger.error('Error during asset download test', 'AssetsManager', err, {
+              asset: { dataObject, distributor },
+            })
           }
         }
       }
 
-      SentryLogger.error('No storage provider was able to provide asset', 'AssetsManager', null, {
-        asset: {
-          contentId,
-          type: resolutionData.assetType,
-          storageProviderIds: storageProvidersToTry.map((sp) => sp.workerId),
-          storageProviderUrls: storageProvidersToTry.map((sp) => sp.metadata),
-        },
+      SentryLogger.error('None of the distributors provided the asset', 'AssetsManager', null, {
+        asset: { dataObject, sortedDistributors },
       })
     })
   }, [
     addAsset,
     addAssetBeingResolved,
     assetIdsBeingResolved,
-    getStorageProviders,
-    getRandomStorageProvider,
+    getAllDistributors,
     pendingAssets,
     removeAssetBeingResolved,
     removePendingAsset,
   ])
 
   return null
 }
+
+// deterministically sort distributors for a given dataObject
+// this is important for caching, if we pick the distributor at random, clients will end up caching [distributors.length] copies of each asset (all have unique URL)
+// TODO: geographical locations into the account, to offer a distributor physically closest to the client, this should ensure best response times
+const sortDistributors = (
+  distributors: DistributorInfo[],
+  dataObject: StorageDataObjectFieldsFragment
+): DistributorInfo[] => {
+  const dataObjectIdBn = new BN(dataObject.id)
+  const distributorsCountBn = new BN(distributors.length)
+  const firstDistributorIndex = dataObjectIdBn.mod(distributorsCountBn).toNumber()
+  return [...distributors.slice(firstDistributorIndex), ...distributors.slice(0, firstDistributorIndex)]
+}
+
+const createDistributorDataObjectUrl = (distributor: DistributorInfo, dataObject: StorageDataObjectFieldsFragment) => {
+  return joinUrlFragments(distributor.endpoint, 'api/v1/assets', dataObject.id)
```

Comment:
Let's move this url to `config/urls.ts`. WDYT?

Reply:
Moved `'api/v1/assets'` to `config/urls`